src/gui/guiChatConsole.cpp

```diff
@@ -639,3 +670,74 @@ void GUIChatConsole::setVisible(bool visible)
 	}
 }
 
+bool GUIChatConsole::isInCtrlKeys(const irr::EKEY_CODE& kc)
+{
+	return kc == KEY_LCONTROL || kc == KEY_RCONTROL || kc == KEY_CONTROL;
+}
+
+void GUIChatConsole::middleClick(s32 col, s32 row)
+{
+	// Prevent accidental rapid clicking
+	static u64 s_oldtime = 0;
+	u64 newtime = porting::getTimeMs();
+
+	// 0.6 seconds should suffice
+	if(newtime - s_oldtime < 600)
+		return;
+	s_oldtime = newtime;
+
+	const std::vector<ChatFormattedFragment> &
+			frags = m_chat_backend->getConsoleBuffer().getFormattedLine(row).fragments;
+	std::string weblink = "";         // from frag meta
+
+	// Identify targetted fragment, if exists
+	int indx = frags.size() - 1;
+	if (indx < 0)
+	{
+		// Invalid row, frags is empty
+		return;
+	}
+	// Minus 1 because the left margin of 1 font space
+	while ((u32)(col - 1) < frags[indx].column)
```

Comment:
Why `u32`, but not `s32` or `s64`? This might have the potential for an integer underflow.

Reply:
That explains some things!